csharp/src/Apache.Arrow.Adbc/C/NativeDelegate.cs

```diff
@@ -20,22 +20,15 @@
 
 namespace Apache.Arrow.Adbc.C
 {
-    internal readonly struct NativeDelegate<T>
+    internal readonly struct NativeDelegate<T> where T : Delegate
     {
         private readonly T _managedDelegate; // For lifetime management
-        private readonly IntPtr _nativePointer;
 
         public NativeDelegate(T managedDelegate)
         {
             _managedDelegate = managedDelegate;
-            _nativePointer = Marshal.GetFunctionPointerForDelegate<T>(managedDelegate);
         }
 
-        public static implicit operator IntPtr(NativeDelegate<T> thunk)
-        {
-            return thunk._nativePointer;
-        }
-
-        public IntPtr Pointer { get { return _nativePointer; } }
+        public IntPtr Pointer { get; }
```

Comment:
Isn't this always a null pointer then?

Reply:
Hmm... not sure what this says about our testing having passed :/. But yes, it looks like I did a poor job copying over from the changed version in the Arrow repo.

Reply:
We don't run C♯ in CI yet here, for what it's worth (unless you mean your local testing)